src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Tasks/LinkerTests.cs

```diff
@@ -70,6 +70,75 @@ static void CreateAbstractIfaceImplementation (string assemblyPath, AssemblyDefi
 			}
 		}
 
+		[Test]
+		public void FixAbstractMethodsStep_NestedTypes ()
+		{
+			var path = Path.Combine (Path.GetFullPath (XABuildPaths.TestOutputDirectory), "temp", TestName);
+			var step = new FixAbstractMethodsStep ();
+			var pipeline = new Pipeline ();
+
+			Directory.CreateDirectory (path);
+
+			using (var context = new LinkContext (pipeline)) {
+
+				context.Resolver.AddSearchDirectory (path);
+
+				var myAssemblyPath = Path.Combine (path, "MyAssembly.dll");
+
+				using (var android = CreateFauxMonoAndroidAssembly ()) {
+					android.Write (Path.Combine (path, "Mono.Android.dll"));
+					CreateNestedInterface (myAssemblyPath, android);
+				}
+
+				using (var assm = context.Resolve (myAssemblyPath)) {
+					step.Process (context);
+
+					var impl = assm.MainModule.GetType ("MyNamespace.Nested/MyClass");
+					var method = impl.Methods.FirstOrDefault (m => m.Name == "MyImplementedMethod");
+					Assert.IsNotNull (method, "MyImplementedMethod should exist");
+					Assert.AreEqual (0, method.Body.Instructions.Count, "MyImplementedMethod should be an empty method");
+					method = impl.Methods.FirstOrDefault (m => m.Name == "MyMissingMethod");
+					Assert.IsNotNull (method, "MyMissingMethod should exist");
+					Assert.AreNotEqual (0, method.Body.Instructions.Count, "MyMissingMethod should *not* be an empty method");
+				}
+			}
+
+			Directory.Delete (path, true);
+		}
+
+		static void CreateNestedInterface (string assemblyPath, AssemblyDefinition android)
+		{
+			using (var assm = AssemblyDefinition.CreateAssembly (new AssemblyNameDefinition ("NestedIFaceTest", new Version ()), "NestedIFaceTest", ModuleKind.Dll)) {
```

Comment:
Possibly "silly" question, but why are we using System.Reflection.Emit to create `MyAssembly.dll` instead of using some C# code?

(Yes, `FixAbstractMethodsStep_NestedTypes()` does that too, but now I wonder the same thing as well!)

Reply:
I tried using CodeDom... It seems like sacrificing too many chickens, it required me adding global alias for System.dll:

https://github.com/jonathanpeppers/xamarin-android/commit/b259e62b144b4ff9439b0fb4878d32b388da638e

I didn't get to the point of how to "add" a new method to an interface of an existing assembly. I guess it would overwrite `Mono.Android.dll` with a new one?

Using only Mono.Cecil it's easy to fabricate the types to make `FixAbstractMethods` do its work.